game/tasks/task05/task_d_the_familiar.py

```diff
@@ -101,25 +122,43 @@ def new(cls) -> "FamiliarSprite":
     @beartype
     def on_update(self, game_clock: GameClock) -> None:
         center = (self.center_x, self.center_y)
-        in_view = self.camera_view.in_view(center)
         # Wait for the user to find the Familiar after reload
-        if not in_view:
+        if not self.camera_view.in_view(center):
             self.change_x, self.change_y = 0, 0
         # For more natural movement, set the Familiar's trajectory to 0 every 0.1s
         elif self.next_update is None or game_clock.current_time > self.next_update:
             self.change_x, self.change_y = 0, 0
             self.next_update = game_clock.get_time_in_future(0.1)
         # Otherwise calculate the trajectory for the Familiar
+
+        # TODO: This class currently needs to understand how each SpriteMotion is
+        #   implemented. The goal for this task is to extract this logic so that
+        #   dependency on motion algorithm is inverted. There are many possible
+        #   solutions and this code currently works as-is, so this is opportunity
+        #   to practice what you've learned so far and think about refactoring a
+        #   challenge like you might see in real code bases.
+        #
+        # TODO: At minimum, you'll want to make this function less complex to resolve
+        #   the warning from flake8 ("R701 - on_update is too complex") that appears
+        #   if you run the "doit run check" task
+        #
+        # As a tip, what kind of abstract interface would you need? What future
+        #   features might be possible that could change the needs for the interface?
         #
-        # TODO: These three conditions are what you will be refactoring
-        #   How can this logic be extracted and specified outside of this
-        #   class without the need for attributes, like 'self.follow'?
+        # There are many possible solutions, so enjoy this final challenge!
         #
-        elif self.follow:
+        elif True:
+            # Go ahead and remove this error once you're ready to start on Task 5,
+            #   this is just to help you find the right place in the code. Good luck!
+            raise NotImplementedError(
+                "Your familiar needs your help!\
+                \nEdit the code in 'task05/task_d_the_familiar.py' to help"
+            )
```

Comment:
What do you think of these changes? I'm logging off, but feel free to make any fixes!

Reply:
Looks good, thanks!